src/app/auth/components/login/login.component.html

```diff
@@ -1,3 +1,15 @@
-<p>
-  login works!
-</p>
+<form #loginForm="ngForm" (ngSubmit)="onSubmit(loginForm)" novalidate>
+  <p>
+    <mat-form-field>
+      <input matInput placeholder="Username" name="username" ngModel required>
+    </mat-form-field>
+  </p>
+  <p>
+    <mat-form-field>
+      <input matInput placeholder="Password" name="password" type="password" ngModel required>
+    </mat-form-field>
+  </p>
+  <p>
+    <button mat-button type="submit" [disabled]="!loginForm.valid">Log in</button>
```

Comment:
a minor thing: we could just negate `loginForm.invalid` instead of `!loginForm.valid`.